api/src/main/java/org/pmiops/workbench/db/dao/UserService.java

```diff
@@ -100,11 +100,30 @@ private User updateWithRetries(Function<User, User> userModifier, User user) {
   }
 
   private void updateDataAccessLevel(User user) {
-    boolean shouldBeRegistered = Optional.ofNullable(user.getIdVerificationIsValid()).orElse(false)
-        && user.getDemographicSurveyCompletionTime() != null
-        && user.getTrainingCompletionTime() != null
-        && user.getTermsOfServiceCompletionTime() != null
+    // TODO: Add in when we add this module
+    // boolean dataUseAgreementCompliant = user.getDataUseAgreementCompletionTime() != null ||
+    // user.getDataUseAgreementBypassTime() != null || !configProvider.get().access.enableDataUseAgreement;
+    Timestamp current = new Timestamp(clock.instant().toEpochMilli());
+    boolean complianceTrainingNotExpired = user.getComplianceTrainingExpirationTime() == null ||
+      current.before(user.getComplianceTrainingExpirationTime()) ||
```

Comment:
Further to my comment above about punting the compliance_training_expiration_time field to a separate PR, I'd simply remove this logic.

Even when we do implement the field rename, I don't think we want to involve the compliance-training expiration time in the logic that determines whether a user should be in the registered group or not. Frankly, we don't quite understand what the "expiration time" we get from Moodle will actually represent. At best, we may use it to drive some user-facing notifications, but even that's uncertain. :)

Plus – for consistency & predictability, all modules should simply check the same core set of fields: completion time, bypass time, and environment-enabled.